packages/features/ee/teams/api/upgrade.ts

```diff
@@ -37,26 +37,27 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
   if (!team) {
     const prevTeam = await prisma.team.findFirstOrThrow({ where: { id } });
     const metadata = teamMetadataSchema.parse(prevTeam.metadata);
+    /** We save the metadata first to prevent duplicate payments */
+    team = await prisma.team.update({
```

Comment:
Update and set team to created team, should not depend on the slug being present.